packages/local_auth/local_auth_darwin/darwin/local_auth_darwin/Sources/local_auth_darwin/LocalAuthPlugin.swift

```diff
@@ -0,0 +1,254 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import Foundation
+import LocalAuthentication
+
+#if canImport(FlutterMacOS)
+  import FlutterMacOS
+#elseif canImport(Flutter)
+  import Flutter
+#endif
+
+class StickyAuthState {
+  let options: AuthOptions
+  let strings: AuthStrings
+  let resultHandler: (Result<AuthResultDetails, Error>) -> Void
+
+  init(
+    options: AuthOptions, strings: AuthStrings,
+    resultHandler: @escaping (Result<AuthResultDetails, Error>) -> Void
+  ) {
+    self.options = options
+    self.strings = strings
+    self.resultHandler = resultHandler
+  }
+}
+
+public final class LocalAuthPlugin: NSObject, FlutterPlugin, LocalAuthApi {
+  private var authContext: AuthContext
+  private let alertController: AlertController
+  private var lastCallState: StickyAuthState?
+
+  init(
+    authContext: AuthContext = DefaultAuthContext(),
+    alertController: AlertController = DefaultAlertController()
+  ) {
+    self.authContext = authContext
+    self.alertController = alertController
+
+    super.init()
+    #if os(iOS)
+      NotificationCenter.default.addObserver(
```

Comment:
I changed it because it's something I see more commonly in Swift, as it's a common pattern for handling application lifecycle events. But I can revert to the previous implementation.

Reply:
it has the same behavior as before, but using NotificationCenter.

Reply:
I understand that you intend it to have the same behavior. What I'm asking is why you are making a change to the logic of the plugin as part of converting the implementation to Swift. I don't see any reason the previous logic would not be trivial to convert to Swift, so I don't understand why this *unrelated* change is being made in a PR that is about changing the language, not the actual implementation.